src/core/matchers/matchersUtil.js

```diff
@@ -167,6 +168,36 @@ getJasmineRequireObj().matchersUtil = function(j$) {
         }
         return result;
       // RegExps are compared by their source patterns and flags.
+      case '[object ArrayBuffer]':
+        // If we have an instance of ArrayBuffer the Uint8Array ctor
+        // will be defined as well
+        var arrayA = new Uint8Array(a);
+        var arrayB = new Uint8Array(b);
+        var arrayALength = arrayA.length;
```

Comment:
I'd like to see the common code paths here pulled out so the existing `Array` code and this code stay in sync.